docs/pages/router/reference/middleware.mdx

```diff
@@ -0,0 +1,200 @@
+
+## Examples
+
+### Authentication
+
+Middleware is often used to perform authorization checks before a route has loaded. You can check headers, cookies, or query parameters to determine if a user has access to certain routes:
+
+```ts app/+middleware.ts
+import { jwtVerify } from 'jose';
+
+export default function middleware(request) {
+  const token = request.headers.get('authorization');
+
+  const decoded = jwtVerify(token, process.env.SECRET_KEY);
+  if (!decoded.payload) {
+    return new Response('Forbidden', { status: 403 });
+  }
+}
+```
+
+### Logging
+
+You can use middleware to log requests for debugging or analytics purposes. This can help you track user activity or diagnose issues in your app:
+
+```ts app/+middleware.ts
+export default function middleware(request) {
+  console.log(`${request.method} ${request.url}`);
+}
+```
+
+### Dynamic redirects
+
+Middleware can also be used to perform dynamic redirects. This allows you to control user navigation based on specific conditions:
+
+```ts app/+middleware.ts
+export default function middleware(request) {
+  if (request.headers.has('specific-header')) {
+    return Response.redirect('https://expo.dev');
+  }
+}
+```
+
+## Additional notes
+
+### Best practices
+
+- Keep middleware lightweight because it runs synchronously on every server request and directly impacts response times.
+- For native apps, use API routes for secure data fetching. When native apps call API routes, those requests will pass through middleware first.
+
+### Typed middleware
+
+```ts app/+middleware.ts
+import { MiddlewareFunction } from 'expo-router';
+
+const middleware: MiddlewareFunction = request => {
+  if (request.headers.has('specific-header')) {
+    return Response.redirect('https://expo.dev');
+  }
+};
+
+export default middleware;
+```
+
+### Limitations
+
+- Middleware runs exclusively on the server and only for HTTP requests. It does not execute during client-side navigation e.g. with [`<Link />`](/versions/latest/sdk/router/#link) or native app screen transitions.
+- The request object passed to middleware is [immutable](#request-immutability) to prevent side effects. You cannot modify headers or consume the request body, ensuring it remains available for route handlers.
+- You can only have one root-level **+middleware.ts** in your app.
+- The same limitations that [apply to API routes](/router/reference/api-routes/#known-limitations) also apply to middleware.
+
+### Request immutability
+
+To prevent unintended side effects and ensure the request body remains available for route handlers, the [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request) passed to middleware is immutable, which means you can:
+
+- Read all request properties like `url`, `method`, `headers`, etc.
+- Read header values using `request.headers.get()`
+- Check for header existence with `request.headers.has()`
+- Access URL parameters and query strings
+
+But you are not allowed to:
```

Comment:
```suggestion
But you may not to:
```

or maybe "you won't be able to" / "can't" / "are prevented from"

Just flagging this because "not allowed to" implies a rule, whereas the code enforces it as fact